gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaaas_driver.py

```diff
@@ -0,0 +1,418 @@
+import socket
+
+from neutron.common import exceptions
+from neutron.common import rpc as n_rpc
+from neutron.db import agents_db
+from neutron.db import agentschedulers_db
+from neutron import manager
+from neutron_vpnaas.db.vpn import vpn_validator
+from neutron_vpnaas.services.vpn.common import constants as const
+from neutron_vpnaas.services.vpn.common import topics
+from neutron_vpnaas.services.vpn.plugin import VPNPlugin
+from neutron_vpnaas.services.vpn import service_drivers
+from neutron_vpnaas.services.vpn.service_drivers import base_ipsec
+
+from oslo_log import log as logging
+import oslo_messaging
+
+LOG = logging.getLogger(__name__)
+
+BASE_IPSEC_VERSION = '1.0'
+
+
+class VPNAgentHostingServiceNotFound(exceptions.NeutronException):
+    message = _("VPN Agent hosting vpn service '%(vpnservice_id)s' not found")
+
+
+class VPNAgentNotFound(exceptions.NeutronException):
+    message = _("VPN Agent not found in agent_db")
+
+
+class IpsecSiteConnOverlappingPeerAdress(exceptions.NeutronException):
+    message = _("Overlapping peer cidr '%(peer_cidr)s' for \
+        peer '%(peer_address)s'")
+
+
+class DuplicateVpnService(exceptions.NeutronException):
+    message = _("Duplicate vpn service '%(name)s'")
+
+
+class VpnServiceInvalidSubnet(exceptions.NeutronException):
+    message = _("Invalid subnet '%(subnet_id)s'")
+
+
+class InvalidVPNServiceDesc(exceptions.NeutronException):
+    message = _("Invalid vpn svc desc '%(desc)s' should be \
+        e.g., fip=192.168.20.120;tunnel_local_cidr=10.0.0.0/24")
+
+
+class VPNServiceInErrorState(exceptions.NeutronException):
+    message = _("VPN service is in error state")
+
```

Comment:
The following classes are unused once OCVpnValidator is removed.
IpsecSiteConnOverlappingPeerAdress
DuplicateVpnService
VpnServiceInvalidSubnet
InvalidVPNServiceDesc
VPNServiceInErrorState